R/get.groundhog.folder.R

```diff
@@ -0,0 +1,133 @@
+#' Get groundhog folder location
+#'
+#' @return the path to the groundhog folder where groundhogR files will be
+#'   stored and where packages loaded with [groundhog.library()] will be
+#'   installed.
+#'
+#' @note you can change the location of this folder by editing the environment
+#'   variable GROUNDHOG_FOLDER. In R, you can do this with the command
+#'   `set.groundhog.folder("path")`.
+#'
+#' @examples
+#' get.groundhog.folder()
+#' @export
+#'
+#' @seealso [set.groundhog.folder()]
+#'
+
+# Function that gets the groundhogR folder, or prompts user to create it.
+get.groundhog.folder <- function() {
+
+  groundhog.folder <- path.expand(Sys.getenv("GROUNDHOG_FOLDER"))
+
+  # If a folder for has not been set, prompt user
+  if (groundhog.folder == "") {
+    # 1. Put the default folder into a variable to show user
+    default.folder <- paste0(path.expand("~"), "/groundhogR/")
+
+    # a=function() {
+    # 2. Show message asking for the desired folder
+    message2(
+      "*****************  Groundhog's directory *********************************************************\n",
+      "<PRESS ENTER> to accept the default directory ('", default.folder, "') for saving packages"
+    )
+    message1(
+      "groundhogR needs a directory to use as a library for saving downloaded and installed packages.\n",
+      "<PRESS ENTER> to use the default for this computer ('", default.folder, "').\n",
+      "Or, type in the folder you would like to use instead (e.g, 'c:/dropbox/groundhogR'). Do not include quotes.\n",
+      "If the folder does not already exists, it will be created.\n",
+      "Type 'quit' to not choose a directory at this time"
+    )
+
+    message2("************************************************************************************************")
+    answer <- readline(prompt = "Please enter your reponse:")
+
+
+    # 3 Quite process if asked to, or asnswer is too short or no slashes
+    # Looks like 'quit'
+    if (tolower(answer) %in% c("quit", "q", "qui", "qit")) {
+      stop("OK folder setting process was stopped.")
+    }
+    # too short
+    if (nchar(answer) < 4 & nchar(answer) > 0) {
+      stop(paste0("The folder name '", answer, "' seems too short to be a folder."))
+    }
+
+    # no slashes
+    slash.count <- regexpr("/", answer)[1] + regexpr("\\\\", answer)[1]
+    if (slash.count < 0 & nchar(answer) > 0) {
+      stop(paste0("The directory name '", answer, "' is missing slashes. It does not seem like a proper directory."))
+    } # End if no slashes
```

Comment:
I don't really like this because again it sounds like we're trying to be too smart and second guessing the users. They might want to use short names or folders named 'qit' and I would finding super annoying that the package prevents me to do so.

@urisohn

Reply:
They can stil have it called qit if they write answer<- '/path/qit'
I am trying to avoid users not reading through and typing, say "OK" , Yes" or "asdsas" without realizing what they are doing.  People who are very comfortable with how R works may make the path short by using the '~'.  I suspect they will be uncomfortable doing that with the ambiguity of how that will be processes and will write down teh full path, but even with ~ we are only preventing single letter folders.

Note that the function set.groundhogr.folder() has not validation, so the message they get could include something like "If you actually meant to name your folder 'git' just run 
`set.goundhogr.folder('git')`

Reply:
I misunderstood the origin of your concern (that's why i deleted my first response to this).
If they want to call it 'qit' they would need to do either 'c:/qit' or '~/qit' or something like that. 

this flexibility, however, became secondary once i added the additional validation of slashes. if they type 'qit' the process will stop anyway. so change to what you feel more comfortable with, as long as we keep the slash check

Reply:
@urisohn, I think keeping even the slash check is a bad idea. It clashes with what every other R function / package does and this kind of unexpected / non-standard behaviour is very non-intuititive and extremely frustating for users in my opinion

Reply:
how about a confirmation 
"you did not include any slashes, you really want this directory?"
It seems very unlikely a user would want to set up a folder with no slashes, no?

Reply:
what scenario are you worried about? where are a user's intentions not granted with this check?

Reply:
Since forcing the presence of slashes is the non-standard behaviour, I think a better question is: what scenario does forcing slashes address?

In my case, I'm simply thinking of the case where the user wants the groundhog folder as a subfolder of their current working directory (e.g., their home). Okay, if it fails on first try, they will do it again with slashes but I've found that this kind of micro-frustrations (which things not working as expected) add up over time and may lead users to give up entirely on a tool when the total frustration reaches a given level.

If you're not convinced after this last comment, I'll follow what you choose.